python/amici/sbml_import.py

```diff
@@ -1202,21 +1227,22 @@ def _replace_in_all_expressions(self,
             for species in self.symbols[SymbolId.SPECIES].values():
                 species['init'] = smart_subs(species['init'],
                                              old, self._make_initial(new))
-                if 'dt' in species:
-                    species['dt'] = smart_subs(species['dt'], old, new)
+                for field in ['dt', 'compartment']:
```

Comment:
I guess some errors in the test suite are due to this, as the compartment symbol is substituted with some value or expression. If this is kept, then maybe the [line in the test script that determines a species' compartment](https://github.com/AMICI-dev/AMICI/blob/fix_infinite_loop/tests/testSBMLSuite.py#L183) can be replaced with [getCompartment](https://github.com/AMICI-dev/AMICI/blob/fix_infinite_loop/tests/testSBMLSuite.py#L189).

Reply:
indeed, that code could and should easily be simplified anyways.